ironfish/src/wallet/scanner/walletScanner.ts

```diff
@@ -177,34 +182,32 @@ export class WalletScanner {
       )
     }
 
-    const connectOnlyAccounts = new Array<Account>()
-    const decryptAndConnectAccounts = new Array<Account>()
-
     for (const candidate of this.scanningAccounts) {
-      if (
-        !candidate.scanFrom ||
-        BufferUtils.equalsNullable(candidate.scanFrom.hash, blockHeader.previousBlockHash)
-      ) {
-        candidate.scanFrom = null
-
-        if (
-          candidate.account.createdAt === null ||
-          blockHeader.sequence >= candidate.account.createdAt.sequence
-        ) {
-          decryptAndConnectAccounts.push(candidate.account)
-        } else {
-          connectOnlyAccounts.push(candidate.account)
+      const { account, scanFrom } = candidate
+
+      if (scanFrom === 'always-scan') {
+        continue
+      }
+
+      if (scanFrom.head === null) {
+        if (account.createdAt === null && blockHeader.sequence === GENESIS_BLOCK_SEQUENCE) {
+          candidate.scanFrom = 'always-scan'
+        }
+
+        if (account.createdAt && blockHeader.sequence >= account.createdAt.sequence) {
+          candidate.scanFrom = 'always-scan'
         }
```

Comment:
Nit: I think this can be simplified as:

```ts
const createdAtSequence = account.createdAt?.sequence ?? GENESIS_BLOCK_SEQUENCE
if (blockHeader.sequence >= createdAtSequence) {
    candidate.scanFrom = 'always-scan'
}
```